src/js/console.js

```diff
@@ -11,45 +19,102 @@ function getConsoleLogs() {
   );
 }
 
-function formattedMessages(messages) {
-  const formattedMessages = messages.map(message => {
-    if (typeof message === 'string') {
-      return message;
-    } else if (message.stack) {
-      return message.stack;
-    } else if (typeof message === 'object') {
-      return JSON.stringify(message);
-    }
+/* function formattedMessages(messages) {
+ const formattedMessages = messages.map(message => {
+   if (typeof message === 'string') {
+     return message;
+   } else if (message.stack) {
+     return message.stack;
+   } else if (typeof message === 'object') {
+     return JSON.stringify(message);
+   }
+   return message;
+ });
+ return formattedMessages;
+}; */
+function formattedMessages(message) {
+  if (typeof message === 'string') {
     return message;
-  });
-  return formattedMessages;
+  }
+  else if (typeof message === 'object') {
+    return JSON.stringify(message);
+  }
 };
 
-console.log = function() {
+console.log = function (arguments) {
   const consoleLogs = getConsoleLogs();
-  consoleLogs.push({ dateTime: new Date(), type: 'log', message: formattedMessages(Array.from(arguments)) });
+  consoleLogs.push({ dateTime: getPSTFromUTC(new Date()), type: 'log', message: formattedMessages(arguments) });
   localStorage.setItem("consoleLogs", JSON.stringify(consoleLogs));
-  console.stdlog.apply(console, arguments);
+  console.stdlog.call(console, arguments);
+
 };
 
-console.debug = function() {
+console.debug = function () {
   const consoleLogs = getConsoleLogs();
-  consoleLogs.push({ dateTime: new Date(), type: 'debug', message: formattedMessages(Array.from(arguments)) });
+  consoleLogs.push({ dateTime: getPSTFromUTC(new Date()), type: 'debug', message: formattedMessages(Array.from(arguments)) });
   localStorage.setItem("consoleLogs", JSON.stringify(consoleLogs));
   console.stdDebug.apply(console, arguments);
 };
 
-console.error = function() {
+console.error = function () {
+  const consoleLogs = getConsoleLogs();
+  consoleLogs.push({ dateTime: getPSTFromUTC(new Date()), type: 'error', message: arguments[1].stack });
```

Comment:
Don't hard code the array index change it back. since the index is based on no of parameters

Reply:
changed as you said